### PR TITLE
Initial project files

### PR DESCRIPTION
**Pull request by austinkelleher**
Using this integration, customers will be able to ingest generic domain information that isn't specific the actual hosting provider that they use. 

A useful sample case would be creating rules that generate alerts for when a domain is going to expire soon etc.

### CODE REVIEW

**Comment by erkangz** [comment]
nice!

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    on_success: never
+    on_failure: always
+
+script: yarn build
+
+deploy:
+  provider: npm
+  email: npm@jupiterone.io
+  api_key:
+    secure: 'travis encrypt <npm@jupiterone.io API key from Dashlane>'
```
Comment (fomentia) [inline .travis.yml]: :thinking:

File: .prettierignore [hunk 2]
```diff
@@ -0,0 +1,3 @@
+coverage
+dist
+.vscode
```
Comment (fomentia) [inline .prettierignore]: Can you add this to your global gitignore instead? I'm not a fan of having editor specific files / extensions in project gitignores.

File: package.json [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        "prettier --write",
+        "git add"
+      ],
+      "*.{js,json}": [
+        "prettier --write",
+        "git add"
+      ]
+    },
+    "ignore": []
+  }
```
Comment (fomentia) [inline package.json]: I understand that these configurations (husky, lint-staged) came from the managed integration example, but I think that they should go into their own respective rc files instead, for consistency with the rest of the configs. I ought to change the example...

File: src/DomainProviderClient.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  lastName: string;
+}
+
+export interface Device {
+  id: string;
+  manufacturer: string;
+  ownerId: string;
+}
+
+export default class DomainProviderClient {
```
Comment (fomentia) [inline src/DomainProviderClient.ts]: Why is this a class? I see that it's initialized and put onto the integration context, but I'm not sure why it's not a function that's called directly.

File: src/types.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import {
+  EntityFromIntegration,
+  GraphClient,
+  IntegrationExecutionContext,
+  PersisterClient,
+} from "@jupiterone/jupiter-managed-integration-sdk";
+import DomainProviderClient from "./DomainProviderClient";
+
+export const DOMAIN_ENTITY_TYPE = "domain1";
```
Comment (fomentia) [inline src/types.ts]: `"domain1"`?

File: tslint.json [hunk 6]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (fomentia) [inline tslint.json]: TSLint is dead, for all new projects we should be using ESLint. I have an example that uses it here: https://github.com/fomentia/open-source-typescript-template

Comment (fomentia) [reply tslint.json]: Also realize this needs to be updated in the integration example... mb :disappointed:

File: .prettierignore (hunk 2 above)
Comment (austinkelleher) [reply .prettierignore]: This came from the managed integration sample.

File: src/types.ts (hunk 5 above)
Comment (austinkelleher) [reply src/types.ts]: Whoops, nice catch. I was testing something and forgot to change back. Will fix.

File: .travis.yml (hunk 1 above)
Comment (austinkelleher) [reply .travis.yml]: Will remove. Result of copying from the managed integration sample.

File: .prettierignore (hunk 2 above)
Comment (fomentia) [reply .prettierignore]: REEEEEEEE that stupid project needs help, i am sorry my son

File: package.json [hunk 7]
```diff
@@ -63,32 +66,6 @@
...
-        "tslint --project .",
-        "prettier --write",
-        "git add"
-      ],
-      "*.{js,json}": [
-        "prettier --write",
-        "git add"
-      ]
-    },
-    "ignore": []
```
Comment (fomentia) [inline package.json]: Sorry if I'm missing something, where did these configs go?

Comment (erkangz) [reply package.json]: added

**Comment by fomentia** [comment]
I don't remember everything in here but your changes seem good. :+1:

**Pull request merged**